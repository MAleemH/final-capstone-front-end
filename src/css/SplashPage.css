* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.list_style_none {
  list-style: none;
}

.td_none {
  text-decoration: none;
}

.land_main_footer {
  padding: 5px 0;
  height: 10vh;
  display: flex;
  gap: 20px;
  text-align: center;
  justify-content: center;
}

.fieldset_border_none {
  border: none;
}

.land_btn1 {
  padding: 10px 8px;
  border-radius: 10px;
  background-color: #d9e1f6;
  border: 1px solid #d9e1f6;
}

.land_btn1 a {
  color: #604b6c;
}
