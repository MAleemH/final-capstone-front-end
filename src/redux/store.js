import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
