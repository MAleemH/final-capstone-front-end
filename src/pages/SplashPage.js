import React from 'react';
import { Link } from 'react-router-dom';
import '../css/SplashPage.css';
import SplashImage from '../img/land_image.jpg';

const SplashPage = () => {
  const SplashImageStyle = {
    backgroundImage: `url(${SplashImage})`,
    backgroundSize: 'contain',
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat',
    width: '100%',
    margin: 'auto',
    height: '85vh',
  };

  return (
    <div className="land_body">

      <main className="land_main" style={SplashImageStyle} />

      <footer className="land_main_footer">
        <fieldset className="fieldset_border_none">
          <button type="button" className="land_btn1">
            {' '}
            <Link to="/login" className="td_none"> LOGIN </Link>
            {' '}
          </button>

        </fieldset>

        <fieldset className="fieldset_border_none">
          <button type="button" className="land_btn1"><Link to="/register" className="td_none"> REGISTER </Link></button>
        </fieldset>
      </footer>
    </div>
  );
};

export default SplashPage;
