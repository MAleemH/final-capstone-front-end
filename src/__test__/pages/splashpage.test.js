import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import SplashPage from '../../pages/SplashPage';

it('RegisterPage component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <SplashPage />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
