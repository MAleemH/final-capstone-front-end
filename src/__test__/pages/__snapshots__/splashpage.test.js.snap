// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage component renders correctly 1`] = `
<div
  className="land_body"
>
  <main
    className="land_main"
    style={
      Object {
        "backgroundImage": "url(land_image.jpg)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "contain",
        "height": "85vh",
        "margin": "auto",
        "width": "100%",
      }
    }
  />
  <footer
    className="land_main_footer"
  >
    <fieldset
      className="fieldset_border_none"
    >
      <button
        className="land_btn1"
        type="button"
      >
         
        <a
          className="td_none"
          href="/login"
          onClick={[Function]}
        >
           LOGIN 
        </a>
         
      </button>
    </fieldset>
    <fieldset
      className="fieldset_border_none"
    >
      <button
        className="land_btn1"
        type="button"
      >
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
           REGISTER 
        </a>
      </button>
    </fieldset>
  </footer>
</div>
`;
